src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/services/utils/TaskMandatoryFieldsValidator.java

```diff
@@ -0,0 +1,150 @@
+                                        @Value("${config.taskMandatoryFields}") List<String> taskMandatoryFields,
+                                        JsonParserUtils jsonParserUtils) {
+        this.launchDarklyFeatureFlagProvider = launchDarklyFeatureFlagProvider;
+        this.isMandatoryFieldCheckEnabled = taskMandatoryFieldCheckEnabled;
+        this.taskMandatoryFields = taskMandatoryFields;
+        this.jsonParserUtils = jsonParserUtils;
+    }
+
+    /**
+     * Validates the mandatory fields of a task.
+     *
+     * @param task the task to be validated
+     */
+    public void validate(TaskResource task) {
+        log.info("Validating mandatory fields for task {}", task.getTaskId());
+
+        if (isMandatoryFieldCheckEnabled) {
+            LDValue mandatoryFieldCheckEnabledServices = launchDarklyFeatureFlagProvider.getJsonValue(
+                FeatureFlag.WA_MANDATORY_FIELD_CHECK,
+                LDValue.of("{\"jurisdictions\": []}")
+            );
+
+            if (mandatoryFieldCheckEnabledServices == null) {
+                log.warn("Mandatory field check enabled services is null, skipping validation.");
+                return;
+            }
+
+            JsonNode excludedJurisdictionsArray =
+                jsonParserUtils.parseJson(mandatoryFieldCheckEnabledServices.toJsonString(),
+                                          MANDATORY_FIELD_CHECK_FLAG_VARIANT.stringValue());
+            if (excludedJurisdictionsArray == null) {
+                log.warn("Excluded jurisdictions array is null, skipping jurisdiction check.");
+                validateTaskMandatoryFields(task);
+                return;
+            }
+
+            if (!isJurisdictionExcluded(task, excludedJurisdictionsArray)) {
+                validateTaskMandatoryFields(task);
+            }
+        }
+    }
+
+    /**
+     * Checks if the task's jurisdiction is excluded.
+     *
+     * @param task the task to be checked
+     * @param excludedJurisdictionsArray the array of excluded jurisdictions
+     * @return true if the jurisdiction is excluded, false otherwise
+     */
+    private boolean isJurisdictionExcluded(TaskResource task, JsonNode excludedJurisdictionsArray) {
+        for (JsonNode jurisdiction : excludedJurisdictionsArray) {
+            if (jurisdiction.asText().equals(task.getJurisdiction())) {
+                log.warn("Task {} is excluded from mandatory field check due to jurisdiction being excluded {}",
+                         task.getTaskId(), task.getJurisdiction());
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Validates the mandatory fields of a task.
+     *
+     * @param task the task to be validated
+     * @throws ValidationException if any mandatory field is missing or invalid
+     * @throws IllegalArgumentException if a property value cannot be found
+     */
+    public void validateTaskMandatoryFields(TaskResource task) {
```

Comment:
Is there a unit test for this?  I can't see anything called TaskMandatoryFieldsValidatorTest only an Integration Test.

Reply:
added one now